bundle/src/main/java/com/adobe/acs/commons/etag/impl/EtagMessageDigestServletFilter.java

```diff
@@ -0,0 +1,178 @@
+        String[] ignoredResponseHeaders() default {"Date", "Cache-Control", "Expires",  "Vary"};
+        
+        @AttributeDefinition(name = "Salt", description = "The (optional) salt is also taken into account for the message digest calculation. It is necessary to change that value whenever the response content or the response headers are now modified differently in a proxy instance between client and AEM (e.g. Dispatcher sets additional headers).")
+        String salt();
+    }
+
+    private static final Logger log = LoggerFactory.getLogger(EtagMessageDigestServletFilter.class);
+
+    private Config configuration;
+    private Collection<String> ignoredHeaderNames;
+
+    @Activate
+    public void activate(Config configuration) {
+        this.configuration = configuration;
+        if (configuration.ignoredResponseHeaders() != null && configuration.ignoredResponseHeaders().length > 0) {
+            ignoredHeaderNames = Arrays.asList(configuration.ignoredResponseHeaders()).stream().map(String::toLowerCase).collect(Collectors.toSet());
+            // turn to lower case
+        } else {
+            ignoredHeaderNames = Collections.emptySet();
+        };
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+       // no-op
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        if (!(response instanceof SlingHttpServletResponse)) {
+            throw new IllegalStateException("Filter not properly registered as Sling Servlet Filter");
+        }
+        if (!configuration.enabled()) {
+            log.debug("ETag filter not enabled");
+            chain.doFilter(request, response);
+        } else {
+            try {
+                // we already checked that this is a HTTP servlet response before
+                SlingHttpServletResponse slingHttpServletResponse = (SlingHttpServletResponse)response;
+                try (BufferedSlingHttpServletResponse bufferedResponse = new BufferedSlingHttpServletResponse((SlingHttpServletResponse) response)) {
+                    chain.doFilter(request, bufferedResponse);
+                    if (!configuration.overwrite() && slingHttpServletResponse.containsHeader(HttpConstants.HEADER_ETAG)) {
+                        log.debug("Do not overwrite existing ETag header with value '{}'", slingHttpServletResponse.getHeader(HttpConstants.HEADER_ETAG));
+                    } else {
+                        if (slingHttpServletResponse.isCommitted()) {
+                            log.error("Can not send ETag header because response is already committed, try to give this filter a higher ranking!");
+                        } else {
+                          String digest = calculateDigestFromResponse(bufferedResponse);
+                          slingHttpServletResponse.setHeader(HttpConstants.HEADER_ETAG, digest);
```

Comment:
Can you log this digest to the request progress tracker as well?

Reply:
Done